AWS-CodePipeline/sam-monorepo-pipeline-template/{{cookiecutter.outputDir}}/codepipeline.yaml

```diff
@@ -0,0 +1,580 @@
+# `sam deploy --config-file pipeline/samconfig-pipeline.toml --config-env pipeline`.
+
+Metadata:
+  cfn-lint:
+    config:
+      ignore_checks:
+        - I3011
+        - I3013
+
+Parameters:
+  ProjectSubfolder:
+    Type: String
+    Default: "{{cookiecutter.monorepo_project_subfolder}}"
+  CodeCommitRepositoryName:
+    Type: String
+    Default: "{{cookiecutter.codecommit_repository_name}}"
+  MainGitBranch:
+    Type: String
+    Default: "{{cookiecutter.main_git_branch}}"
+  MonorepoSsmPrefix:
+    Type: String
+    Default: "{{cookiecutter.monorepo_ssm_prefix}}"
+  # Stage 1 Parameters
+  TestConfigEnvName:
+    Type: String
+    Default: "{{cookiecutter.testing_stage_name}}"
+  TestingPipelineExecutionRole:
+    Type: String
+    Default: "{{cookiecutter.testing_pipeline_execution_role}}"
+  TestingCloudFormationExecutionRole:
+    Type: String
+    Default: "{{cookiecutter.testing_cloudformation_execution_role}}"
+  # Stage 2 Parameters
+  ProdConfigEnvName:
+    Type: String
+    Default: "{{cookiecutter.prod_stage_name}}"
+  ProdPipelineExecutionRole:
+    Type: String
+    Default: "{{cookiecutter.prod_pipeline_execution_role}}"
+  ProdCloudFormationExecutionExeRole:
+    Type: String
+    Default: "{{cookiecutter.prod_cloudformation_execution_role}}"
+
+Resources:
+  #  ____  _            _ _
+  # |  _ \(_)_ __   ___| (_)_ __   ___
+  # | |_) | | '_ \ / _ | | | '_ \ / _ \
+  # |  __/| | |_) |  __| | | | | |  __/
+  # |_|   |_| .__/ \___|_|_|_| |_|\___|
+  #         |_|
+  Pipeline:
+    Type: AWS::CodePipeline::Pipeline
+    Properties:
+      ArtifactStore:
+        Location: !Ref PipelineArtifactsBucket
+        Type: S3
+      RoleArn: !GetAtt CodePipelineExecutionRole.Arn
+      RestartExecutionOnUpdate: false
+      Stages:
+        - Name: Source
+          Actions:
+            - Name: SourceCodeRepo
+              ActionTypeId:
+                Version: "1"
+                Category: Source
+                Owner: AWS
+                Provider: CodeCommit
+              Configuration:
+                BranchName: !Ref MainGitBranch
+                RepositoryName: !Ref CodeCommitRepositoryName
+                PollForSourceChanges: false
+              OutputArtifacts:
+                - Name: SourceCodeAsZip
+              RunOrder: 1
+
+        - Name: UpdatePipeline
+          Actions:
+            - Name: CreateChangeSet
+              ActionTypeId:
+                Category: Deploy
+                Owner: AWS
+                Provider: CloudFormation
+                Version: "1"
+              Configuration:
+                ActionMode: CHANGE_SET_REPLACE
+                RoleArn: !GetAtt PipelineStackCloudFormationExecutionRole.Arn
+                StackName: !Ref AWS::StackName
+                ChangeSetName: !Sub ${AWS::StackName}-ChangeSet
+                TemplatePath: !Sub "SourceCodeAsZip::${ProjectSubfolder}/codepipeline.yaml"
```

Comment:
This will work if the lambda function's source code is in its own file. It will need to use `sam deploy` instead of `cfn deploy` this is used. Maybe let's move the lambda function back to template for now 😢, and we can do that in a follow up.

Reply:
Sure, happy w/ that.

Reply:
Done.